src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<Machine> {
+    @Override
+    public List getAll(Class type) {
+        MachineProducer machineProducer;
+        if (type == Truck.class) {
+            machineProducer = new TruckProducer();
+        } else if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
+        } else if (type == Excavator.class) {
+            machineProducer = new ExcavatorProducer();
+        } else {
+            return new ArrayList();
+        }
+        return (List<Machine>) machineProducer.get();
+    }
+
+    @Override
+    public void fill(List machines, Machine value) {
```

Comment:
```suggestion
    public void fill(List<....> machines, Machine value) {
```
аналогічно в startWorking